src/pulselistener/pulselistener/mercurial.py

```diff
@@ -192,13 +189,10 @@ class MercurialWorker(object):
     '''
     Mercurial worker maintaining a local clone of mozilla-unified
     '''
-    def __init__(self, phabricator_api, publish_phabricator, cache_root, repositories):
+    def __init__(self, queue_name, queue_phabricator, phabricator_api, cache_root, repositories):
+        self.queue_name = queue_name
+        self.queue_phabricator = queue_phabricator
         self.phabricator_api = phabricator_api
-        self.publish_phabricator = publish_phabricator
-        logger.info('Phabricator publication is {}'.format(self.publish_phabricator and 'enabled' or 'disabled'))  # noqa
-
-        # Build asyncio shared queue
-        self.queue = asyncio.Queue()
 
         # Configure repositories, and index them by phid
         self.repositories = {
```

Comment:
At some point we should probably totally remove the phabricator API usage from Mercurial